components/x-gift-article/src/GiftArticle.jsx

```diff
@@ -188,6 +193,13 @@ const withGiftFormActions = withActions(
 						}
 					}
 				}
+			},
+			includeHighlightsHandler() {
+				return (state) => {
+					const includeHighlights = !state.includeHighlights
+					initialProps.includeHighlights = includeHighlights
+					return { includeHighlights }
+				}
```

Comment:
why is this action changing a property in initialProps? this doesn't look like a good pattern to me, shouldn't it only update the state?